### DISCUSSION TITLE
Static discovery for non IP transport types

### DISCUSSION THREAD

**Discussion by valerijbielskij** [OpenDDS/OpenDDS#3413] [Q&A] [answered]
Hi,

we are planning to use OpenDDS in an environment where IP stack is not available. First we will implement custom transport protocol then we will have to figure out how to deal with the discovery. We would very much like to use built-in static discovery mechanism but unfortunately OpenDDS Developer's guide states the following - "Currently, static discovery can only be used for endpoints using the RTPS UDP transport.". My questions is what's the rationale behind this limitation and how much effort would it require to extend it to support more (all?) transport types. Any hints or guidelines are appreciated.

**Comment by mitza-oci** [comment, marked as answer]
It would depend on the use case, but instead of looking at static discovery I'd suggest looking at RTPS Discovery and replacing the parts that depend on UDP/IP sockets and the rtps_udp transport.  Static discovery is very limited in terms of functionality.

**Reply by valerijbielskij** [reply]
Thanks for the reply. The reason I'm trying to avoid RTPS discovery is because it appears as we don't need to use SEDP, SPDP can probably be also avoided. First of all system will be constrained on the resources, besides we need to minimize the traffic as much as possible. In our case domains, endpoints, topics and even QoS configs are all static and known during the initialization of services. In the future we might need to support dynamic updates of basic QoS parameter like reliability kind but even then the same value would be applied for all existing participants, therefore it would eventually become consistent. Now knowing a bit more details would you still insist on choosing RTPS discovery?

**Reply by mitza-oci** [reply]
I'm not insisting on anything, it's open source so you can do what you'd like.
Thanks for providing some additional context.
I'm hesitant to recommend static discovery since it is a very limited subset of what we typically consider a "full DDS" system -- no DDS Security, XTypes, some QoS policies probably won't work (not just changing the QoS, but the functionality of the policy itself).
You might consider doing a prototype of the distributed application based on the rtps_udp transport and static discovery.  Make sure that the prototype gives you the features and behavior you need.  Then you could move it to your own transport.  There are some functions of the rtps_udp transport that only exist to support static discovery -- those would have to be implemented in your new transport.

**Comment by valerijbielskij** [comment]
Pardon me for using a wrong word, indeed I meant to use recommend instead of insist. Our use case is a bit special (and maybe also dubious) in regards of DDS functionality, we want to use DDS as a transport system to build a distribution platform so it's unlikely that we will need XTypes or full QoS support, DDS Security is on the agenda though but only in the far future. Your comments are very appreciated, I will try out the suggestions, thanks a lot.
